src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/blocks/HologramProjector.java

```diff
@@ -65,16 +78,30 @@ public BlockUseHandler onRightClick() {
             Player p = e.getPlayer();
             Block b = e.getClickedBlock().get();
 
+            // Updates old holograms to use new coordinate system
+            if (BlockStorage.getLocationInfo(b.getLocation(), Y_OFFSET_PARAMETER) == null) {
+                BlockStorage.addBlockInfo(b, X_OFFSET_PARAMETER, "0.5");
+                BlockStorage.addBlockInfo(b, Y_OFFSET_PARAMETER, BlockStorage.getLocationInfo(b.getLocation(), "offset"));
+                BlockStorage.addBlockInfo(b, Z_OFFSET_PARAMETER, "0.5");
+            }
+
             if (BlockStorage.getLocationInfo(b.getLocation(), "owner").equals(p.getUniqueId().toString())) {
                 openEditor(p, b);
+            } else {
+                SlimefunPlugin.getLocalization().sendMessage(p, "no-permission", true);
             }
         };
     }
 
-    private static void openEditor(Player p, Block projector) {
+    private void openEditor(Player p, Block projector) {
         ChestMenu menu = new ChestMenu(SlimefunPlugin.getLocalization().getMessage(p, "machines.HOLOGRAM_PROJECTOR.inventory-title"));
 
-        menu.addItem(0, new CustomItem(Material.NAME_TAG, "&7Text &e(Click to edit)", "", "&r" + ChatColors.color(BlockStorage.getLocationInfo(projector.getLocation(), "text"))));
+        for (int s : BORDER) {
+            menu.addItem(s, new CustomItem(ChestMenuUtils.getBackground(), " "));
+            menu.addMenuClickHandler(s, ChestMenuUtils.getEmptyClickHandler());
+        }
+
+        menu.addItem(0, new CustomItem(Material.NAME_TAG, "&7Text &e(Click to edit)", "", ChatColor.WHITE + ChatColors.color(BlockStorage.getLocationInfo(projector.getLocation(), "text"))));
```

Comment:
You could do this to make it a bit more readable.
```suggestion
        String text = ChatColor.WHITE + ChatColors.color(BlockStorage.getLocationInfo(projector.getLocation(), "text"));
        menu.addItem(0, new CustomItem(Material.NAME_TAG, "&7Text &e(Click to edit)", "", text));
```